### DISCUSSION TITLE
How to fix custom component keyboard cursor behavior

### DISCUSSION THREAD

**Discussion by icyxianyu** [ueberdosis/tiptap#5898] [General Questions]
I tried to implement a custom extension to achieve behavior similar to an inline input box with a placeholder.
<img width="534" alt="image" src="https://github.com/user-attachments/assets/fb7aa09a-9cfa-4a53-90cd-d5c16f6cebf9">

 However, when I tried to switch left and right through the keyboard, I couldn't get the cursor to enter the custom extension，custom components will be selected for all ,At the same time, I also found that rows with custom extensions had abnormal behavior to switch up and down. 

![录屏2024-12-01 15 02 23](https://github.com/user-attachments/assets/b34efa4c-210a-4ece-9111-5b413fc15f9e)

here is a demo: https://codesandbox.io/p/sandbox/hrjg4g

How can I fix it ? Or is there another implementation?

**Comment by bdbch** [comment]
Lets keep this in the issue you created #5903
